webapp/src/js/services/db-sync.js

```diff
@@ -77,6 +131,9 @@ angular
               // TODO: work out what `err` looks like and navigate to login
               // when we detect it's a 401
               $log.error(`Denied replicating ${direction.name} remote server`, err);
+              if (direction.onDenied) {
```

Comment:
We should probably do the above TODO at the same time - no point retrying if you're not logged in...